### DISCUSSION TITLE
Building from CLI

### DISCUSSION THREAD

**Discussion by bdbell** [grblHAL/core#372] [Q&A]
I am attempting to build a GrblHAL system from a headless computer.  Is there any documented way to build without needing an IDE?  I noticed a few drivers support PlatformIO but most seem to still need an IDE.  I am aware of the web builder but would still like to explore the local build process.

**Comment by terjeio** [comment]
If PlatformIO is supported for the processor and framework then this would be the easiest to set up?

> Is there any documented way to build without needing an IDE?

I have not made any documentation for CLI based builds, and I do not think I will as I simply do not have free time to do so. If someone else is willing to do it that would be nice though. E.g. all current  PlatformIO instructions and .ini-files has been provided by users, I am grateful for that as I am not up to speed on how to set up PlatformIO myself (even if I use it for most of the Web Builder builds).

>  I noticed a few drivers support PlatformIO but most seem to still need an IDE.

I believe all can be build from a CLI provided that the neccesary toolchain, framework and makefiles are set up correctly since this is what an IDE installation provides. If the build system is based on cmake then it may not be too hard to get [CLI based builds up and running](https://forums.raspberrypi.com/viewtopic.php?t=356141). If not then likely harder as makefile(s) needs to be generated if the build is not a static one (a single configuration). Anyway, I am not an expert on this topic as the IDEs I use (mainly Eclipse based) hides the build process from me.

---

What could perhaps made a bit clearer by me is that builds can be configured by command line parameters as nearly all configuration #defines can be overridden - which the Web Builder takes advantage of.
